lib/system.nim

```diff
@@ -1571,19 +1571,33 @@ proc isNil*[T](x: seq[T]): bool {.noSideEffect, magic: "IsNil", nilError.}
   ## * `isNil(string) <#isNil,string>`_
 
 proc isNil*[T](x: ref T): bool {.noSideEffect, magic: "IsNil".}
+  ## `x == nil` or `x != nil` is preferred, please use that instead.
+  # see #16191
```

Comment:
1. Please, don't adopt this Timothee-style of (pointless) references.
2. What exactly is there to see in #16191 (this PR)?